server/tests/user-service.test.ts

```diff
@@ -0,0 +1,128 @@
+import { User } from '../entities/user';
+import { mock, instance, when, verify, deepEqual, resetCalls } from 'ts-mockito';
+import { UserService } from '../service/user-service';
+import { Dependencies } from '../types';
+import { Repository, DeleteResult, UpdateResult } from 'typeorm';
+
+const mockRepository = <Repository<User>>mock(Repository);
+const repoInstance = instance(mockRepository);
+const service = new UserService(<Dependencies>{ userRepository: repoInstance });
+const testUser = new User('three');
+const users: User[] = [new User('one'), new User('two'), testUser];
+
+beforeEach(() => {
+  resetCalls(mockRepository);
```

Comment:
Looks like solids tests 💪  One thing I was thinking should we use `reset` instead of `resetCalls`? Because `reset` resets also mocked function return aka when you use `when`. What do you think?

Reply:
Hmm, I suppose the `reset` options is actually the correct one here as each test decalres the stubs used anyway, so if anything, using `reset` makes sure some random side effect from another test isn't the reason why something else passes.